packages/smooth_app/lib/pages/scan/scan_page.dart

```diff
@@ -32,13 +35,25 @@ class _ScanPageState extends State<ScanPage> {
 
   @override
   Widget build(BuildContext context) {
+    final UserPreferences userPreferences = context.read<UserPreferences>();
     if (_model == null) {
       return const Center(child: CircularProgressIndicator());
     }
 
+    final Widget child;
+
+    if (userPreferences.getFlag(
+          UserPreferencesDevMode.userPreferencesFlagUseMLKit,
+        ) ??
+        true) {
+      child = const MLKitScannerPage();
+    } else {
+      child = const ContinuousScanPage();
+    }
```

Comment:
I don't know how intertwined the Widgets are, but it would have been cleaner to use the same page and inside the same page use a different Widget.

Reply:
The widget tree is mostly the same, we could extract this, but since most of it is already in the helper there isn't probably much to edit here, since both use a totally different way of setting up